prql-compiler/src/sql/anchor.rs

```diff
@@ -88,20 +88,19 @@ impl AnchorContext {
         cids.iter().map(|cid| self.materialize_expr(cid)).collect()
     }
 
-    pub fn materialize_name(&mut self, cid: &CId) -> String {
+    pub fn materialize_name(&mut self, cid: &CId) -> (Option<String>, String) {
         // TODO: figure out which columns need name and call ensure_column_name in advance
         // let col_name = self
         //     .get_column_name(cid)
         //     .expect("a column is referred by name, but it doesn't have one");
         let col_name = self.ensure_column_name(cid);
 
-        if let Some(tid) = self.columns_loc.get(cid) {
+        let table = self.columns_loc.get(cid).map(|tid| {
             let table = self.table_defs.get(tid).unwrap();
 
-            format!("{}.{}", table.name, col_name)
-        } else {
-            col_name
-        }
+            table.name.clone()
+        });
+        (table, col_name)
     }
```

Comment:
Here, materialize_name was concatinating table name and var name. We separated them for the whole pipeline, but merged them together as a last step before split them back up in `translate_ident`.

Reply:
Much nicer!

(also fine to point me towards things like this, also fine to DIY like this tho)

Reply:
Yeah, sorry I took over your PR :D I was excited to see it mostly done, because I knew it composes well with other parts of translator.

Reply:
No prob at all — thanks for finishing it!